File: 6_kyu/Find_The_Parity_Outlier.java
public class FindOutlier {
    static int find(int[] integers) {
      int cuentaPar = 0;
      int par = 0;
      int impar = 0;
  
      for (int i : integers) {
        if (i % 2 == 0) {
          cuentaPar++;
          par = i;
        } else {
          impar = i;
        }
      }
  
      return cuentaPar == 1 ? par : impar;
  
    }
  }

File: 6_kyu/Find_The_Parity_Outlier.php
<?php

function find($integers) {
    //Obtener un array de pares
    $pares = array_filter($integers, function ($valor) { return $valor % 2 === 0;});
    //Se obtiene el array de impares solo si es necesario, sino se devuelve el valor del unico par.
    return sizeof($pares) === 1 ? reset($pares) : reset(array_diff($integers, $pares));
}